### PR TITLE
MVC Intro tutorial .NET 7 update

### LINKED ISSUE #27188: Update MVC intro tutorial for .NET 7
---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: faf18d20-d28f-1efa-d329-181fff3630fa
* Version Independent ID: 46a7c2e0-4871-491f-02c2-54f7df2237b1
* Content: [Get started with ASP.NET Core MVC](https://learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/start-mvc?view=aspnetcore-6.0&tabs=visual-studio)
* Content Source: [aspnetcore/tutorials/first-mvc-app/start-mvc.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/tutorials/first-mvc-app/start-mvc.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #27188 

[Internal review URL](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/start-mvc?view=aspnetcore-7.0&branch=pr-en-us-27268&tabs=visual-studio)

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
Under [Run the app](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/start-mvc?view=aspnetcore-7.0&branch=pr-en-us-27268&tabs=visual-studio-code#run-the-app-3), I'd move Trust Cert first and perhaps have it under it's own H3. Right now the instructions are too long and mixed up. I'm only looking at this because I'm copying what you have in the Web API tutorial.

BTW, it doesn't navigate to `https://localhost:5001`

**Comment by tdykstra** [comment]
>move Trust Cert first and perhaps have it under it's own H3

In the VS instructions it can't go first as the include is currently written, because it relies on Ctrl+F5 to initiate the trust sequence. Also, for both VS and VSC the person doing the tutorial might already have a trusted cert, so it makes sense to start with Ctrl+F5 rather than "Trust the certificate". What's missing in the VSC instructions is something that tells you what happens when you try to run the app and you don't have a trusted cert.

**Comment by Rick-Anderson** [comment]
> In the VS instructions it can't go first as the include is currently written, because it relies on Ctrl+F5 to initiate the trust sequence.
It can go first and it  doesn't matter if it's already trusted.

> Also, for both VS and VSC the person doing the tutorial might already have a trusted cert, so it makes sense to start with Ctrl+F5 rather than "Trust the certificate".

Why? Here's what it says when you run it thousands of time with a currently trusted cert

>dotnet dev-certs https --trust
Trusting the HTTPS development certificate was requested. A confirmation prompt will be displayed if the certificate was not previously trusted. Click yes on the prompt to trust the certificate.
Successfully trusted the existing HTTPS certificate.

You just don't get prompted.


 What's missing in the VSC instructions is something that tells you what happens when you try to run the app and you don't have a trusted cert.

I find the instructions way to complicated and long compared to the Web API instructions

**Comment by tdykstra** [comment]
I don't see what you're looking at that is simpler -- they all (MVC, web api, minimal web api) use the same include file for the cert trust steps.

**Comment by Rick-Anderson** [comment]
Include needs to be fixed
image

File: aspnetcore/tutorials/first-mvc-app/start-mvc/sample/7.0-completed/Controllers/HelloWorld.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+using Microsoft.AspNetCore.Mvc;
+using System.Text.Encodings.Web;
+
+namespace MvcMovie.Controllers
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/first-mvc-app/start-mvc/sample/7.0-completed/Controllers/HelloWorld.cs:4]:
```suggestion
namespace MvcMovie.Controllers;
```

File: aspnetcore/tutorials/first-mvc-app/start-mvc/sample/7.0-completed/Models/Movie.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace MvcMovie.Models
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/first-mvc-app/start-mvc/sample/7.0-completed/Models/Movie.cs:5]:
```suggestion
namespace MvcMovie.Models;
```

File: aspnetcore/tutorials/first-mvc-app/start-mvc.md [hunk 3]
```diff
@@ -599,10 +599,12 @@ For Visual Studio for Mac, see the .NET 5 version of this tutorial.
 * In the **Create a new project** dialog, select **ASP.NET Core Web App (Model-View-Controller)** > **Next**.
 * In the **Configure your new project** dialog, enter `MvcMovie` for **Project name**. It's important to name the project *MvcMovie*. Capitalization needs to match each `namespace` when code is copied.
 * Select **Next**.
-* In the **Additional information** dialog, select **.NET 6.0 (Long-term support)**.
+* In the **Additional information** dialog:
+  * Select **.NET 7.0**.
+  * Uncheck **Do not use top-level statements**
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/first-mvc-app/start-mvc.md:604]: Shouldn't this be
Verify **Do not use top-level statements** is unchecked?
Isn't that the default?

File: aspnetcore/tutorials/first-mvc-app/start-mvc.md [hunk 4]
```diff
@@ -599,10 +599,12 @@ For Visual Studio for Mac, see the .NET 5 version of this tutorial.
...
 * Select **Create**.
 
-![Additional info dialog](start-mvc/_static/net6-additional-info.png)
+![Additional info dialog](start-mvc/_static/net7-additional-info.png)
 
 For alternative approaches to create the project, see [Create a new project in Visual Studio](/visualstudio/ide/create-new-project).
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/first-mvc-app/start-mvc.md:609]:
```suggestion
For more information including alternative approaches to create the project, see [Create a new project in Visual Studio](/visualstudio/ide/create-new-project).
```

**Comment by Rick-Anderson** [comment]
Probably too late now, but I'm putting moniker 7 at the top, not the bottom.

File: aspnetcore/tutorials/first-mvc-app/start-mvc/sample/7.0-completed/Controllers/HelloWorld.cs (hunk 1 above)
Comment (tdykstra) [reply aspnetcore/tutorials/first-mvc-app/start-mvc/sample/7.0-completed/Controllers/HelloWorld.cs:4]: I thought of switching to file-scoped namespaces, but the generated code still uses old style, so it seemed better to leave other cases of old style alone so everything would be consistent.

Comment (Rick-Anderson) [reply aspnetcore/tutorials/first-mvc-app/start-mvc/sample/7.0-completed/Controllers/HelloWorld.cs:4]: We've been using file-scoped for all code we generate or even templated generated when we make substantial changes. It looks much better. The templates should catch up soon, we should lead the way. 

@guardrex, are you using file scoped namespaces or 1990's style?

Comment (guardrex) [reply aspnetcore/tutorials/first-mvc-app/start-mvc/sample/7.0-completed/Controllers/HelloWorld.cs:4]: Blazor templates started using them in 6.0; so insofar as I used template code, I started back then. However, I only started converting other examples recently.

I wish that they had gone a different way with line placement to save another vertical line ... https://github.com/dotnet/aspnetcore/issues/44676 ... but they said *'Nope!'* 😢

File: aspnetcore/tutorials/first-mvc-app/start-mvc.md (hunk 3 above)
Comment (tdykstra) [reply aspnetcore/tutorials/first-mvc-app/start-mvc.md:604]: If they already selected it for a different project earlier, the default changes to selected, so it's safer to just say uncheck it.

Comment (Rick-Anderson) [reply aspnetcore/tutorials/first-mvc-app/start-mvc.md:604]: We'll see which version has more confusion, my RP verify vs your action verb.  Unlikely they'll have done it before, and if they have, verify takes care of it.

File: aspnetcore/tutorials/first-mvc-app/start-mvc.md [hunk 5]
```diff
@@ -599,12 +599,14 @@ For Visual Studio for Mac, see the .NET 5 version of this tutorial.
 * In the **Create a new project** dialog, select **ASP.NET Core Web App (Model-View-Controller)** > **Next**.
 * In the **Configure your new project** dialog, enter `MvcMovie` for **Project name**. It's important to name the project *MvcMovie*. Capitalization needs to match each `namespace` when code is copied.
 * Select **Next**.
-* In the **Additional information** dialog, select **.NET 6.0 (Long-term support)**.
+* In the **Additional information** dialog:
+  * Select **.NET 7.0**.
+  * Uncheck **Do not use top-level statements**
```
Comment (tdykstra) [inline aspnetcore/tutorials/first-mvc-app/start-mvc.md:604]:
```suggestion
Verify that **Do not use top-level statements** is unchecked.
```

**Pull request merged**